src/Symfony/Bundle/SecurityBundle/Templating/Helper/LogoutUrlHelper.php

```diff
@@ -88,15 +103,27 @@ public function getLogoutUrl($key)
     /**
      * Generates the logout URL for the firewall.
      *
-     * @param string      $key           The firewall key
+     * @param string|null $key           The firewall key or null to use the current firewall key
      * @param bool|string $referenceType The type of reference (one of the constants in UrlGeneratorInterface)
      *
      * @return string The logout URL
      *
-     * @throws \InvalidArgumentException if no LogoutListener is registered for the key
+     * @throws \InvalidArgumentException if no LogoutListener is registered for the key or the key could not be found automatically.
      */
     private function generateLogoutUrl($key, $referenceType)
     {
+        // Fetch the current provider key from token, if possible
+        if (null === $key && $this->tokenStorage) {
```

Comment:
I would be more explicit when checking the `tokenStorage`: `null !== $this->tokenStorage`